src/window_manager.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+#ifndef WINDOW_MANAGER_H
+#define WINDOW_MANAGER_H
+
+#include <vector>
+#include <string>
+#include "types.h"
+
+std::vector<WindowHandle> getWindows();
+WindowHandle getActiveWindow();
+
+std::string getWindowTitle(const WindowHandle windowHandle);
+MMRect getWindowRect(const WindowHandle windowHandle);
```

Comment:
It feels like some documentation would make sense to provide an overview of the functionality provided by the different implementations.